freqtrade/freqtradebot.py

```diff
@@ -821,10 +825,13 @@ def cancel_stoploss_on_exchange(self, trade: Trade) -> Trade:
         return trade
 
     def get_valid_enter_price_and_stake(
-            self, pair: str, price: Optional[float], stake_amount: float,
-            side: str, trade_side: str,
-            entry_tag: Optional[str],
-            trade: Optional[Trade]) -> Tuple[float, float]:
+        self, pair: str, price: Optional[float], stake_amount: float,
+        side: str, trade_side: str,
+        entry_tag: Optional[str],
+        trade: Optional[Trade]
+    ) -> Tuple[float, float]:
+        # TODO: This method has no tests
```

Comment:
I'm not sure about this ...
it might not have explicit tests, only testing this behaviour - but the content itself is tested as part of execute_entry